src/client.rs

```diff
@@ -71,81 +22,28 @@ pub(crate) fn check_key_len(key: &str) -> Result<(), MemcacheError> {
 }
 
 impl Client {
-    #[deprecated(since = "0.10.0", note = "please use `connect` instead")]
-    pub fn new<C: Connectable>(target: C) -> Result<Self, MemcacheError> {
-        return Self::connect(target);
+    pub fn with_pool(pool: Pool<ConnectionManager>) -> Self {
+        Self(pool)
     }
 
-    pub fn with_pool_size<C: Connectable>(target: C, size: u32) -> Result<Self, MemcacheError> {
-        let urls = target.get_urls();
-        let mut connections = vec![];
-        for url in urls {
-            let parsed = Url::parse(url.as_str())?;
-            let pool = r2d2::Pool::builder()
-                .max_size(size)
-                .build(ConnectionManager::new(parsed))?;
-            connections.push(pool);
-        }
-        Ok(Client {
-            connections,
-            hash_function: default_hash_function,
-        })
-    }
+    /// Connect to memcache via given target Url
+    pub fn connect(target: impl AsRef<str>) -> Result<Self, MemcacheError> {
+        let pool = r2d2::Pool::builder()
+            .max_size(20)
```

Comment:
shouldn't size be an argument?

Reply:
I was thinking about sane value and tradeoffs of exposing this as argument :thinking:, maybe compromise would be connnect_with_size ? on the other hand what about the connection_timeout :smile:

Reply:
Yeah, the rust way of doing things would be probably exposing another method such as `connnect_with_size` or creating a builder.

Reply:
Left only wayt to initialize memcache `with_pool` function